erpnext/setup/doctype/territory/territory.js

```diff
@@ -11,29 +11,23 @@ frappe.ui.form.on("Territory", {
 				}
 			}
 		};
+		frm.fields_dict['parent_territory'].get_query = function(doc) {
+			return{
+				filters:[
+					['Territory', 'is_group', '=', 1],
+					['Territory', 'name', "!=", doc.territory_name]
+				]
+			}
+		}
+	},
+	refresh: function(frm) {
+		frm.trigger("set_root_readonly");
+	},
+	set_root_readonly: function(frm) {
```

Comment:
@vishdha we must make this a common function in utils maybe, the code seems repetative.